modules/web/src/app/kubeone-wizard/steps/credentials/provider/basic/aws/component.ts

```diff
@@ -72,12 +77,32 @@ export class KubeOneAWSCredentialsBasicComponent extends BaseFormValidator imple
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(_ => this.form.reset());
 
+    this.form.valueChanges
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(_ =>
+        this._presetsService.enablePresets(Object.values(Controls).every(control => !this.form.get(control).value))
+      );
+
+    this._presetsService.presetChanges
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe(preset => Object.values(Controls).forEach(control => this._enable(!preset, control)));
+
     merge(this.form.get(Controls.AccessKeyID).valueChanges, this.form.get(Controls.AccessKeySecret).valueChanges)
       .pipe(distinctUntilChanged())
       .pipe(takeUntil(this._unsubscribe))
       .subscribe(_ => (this._clusterSpecService.cluster = this._getClusterEntity()));
   }
 
+  private _enable(enable: boolean, name: string): void {
```

Comment:
```suggestion
  private _enable(enable: boolean, name: Controls): void {
```

Reply:
so this method been used a lot in every where isn't better to use a generic method, or it's too late for that ?

Reply:
Yea, at some point we'll need to move this to some shared place.